shared/src/commonMain/kotlin/io/customer/shared/tracking/api/HttpClientBuilder.kt

```diff
@@ -0,0 +1,79 @@
+package io.customer.shared.tracking.api
+
+import io.customer.shared.device.UserAgentStore
+import io.customer.shared.sdk.config.NetworkConfig
+import io.customer.shared.sdk.meta.Region
+import io.customer.shared.sdk.meta.Workspace
+import io.customer.shared.util.LogLevel
+import io.customer.shared.util.Logger
+import io.ktor.util.*
+
+/**
+ * Helper class to provide http client params from a single place.
+ */
+internal interface HttpClientBuilder {
+    val requestTimeoutMillis: Long
+    val clientLogLevel: io.ktor.client.plugins.logging.LogLevel
+    val clientLogger: io.ktor.client.plugins.logging.Logger
+
+    /**
+     * Returns trackingApiUrl provided in config; fallbacks to tracking url from [Region].
+     */
+    val baseURL: String
+
+    /**
+     * The headers provided will be automatically included in all HTTP requests made using the
+     * client.
+     */
+    val headers: Map<String, String>
+}
```

Comment:
Kept these attributes in a separate class as I thought it would be easy to write tests for these properties this way.